pkg/shardddl/pessimism/keeper.go

```diff
@@ -16,26 +16,30 @@ package pessimism
 import (
 	"sync"
 
+	"go.etcd.io/etcd/clientv3"
+
 	"github.com/pingcap/dm/dm/master/metrics"
 	"github.com/pingcap/dm/pkg/utils"
 )
 
 // LockKeeper used to keep and handle DDL lock conveniently.
 // The lock information do not need to be persistent, and can be re-constructed from the shard DDL info.
 type LockKeeper struct {
-	mu    sync.RWMutex
-	locks map[string]*Lock // lockID -> Lock
+	mu             sync.RWMutex
+	locks          map[string]*Lock                          // lockID -> Lock
+	latestDoneDDLs map[string]map[string]map[string][]string // task -> downSchema -> downTable -> ddls
```

Comment:
seems

```
		lockID           = fmt.Sprintf("%s-`%s`.`%s`", task, downSchema, downTable)
```

and `latestDoneDDLs` should be a part of a `Lock`

Reply:
Oh, i'm reverting https://github.com/pingcap/dm/pull/1626/commits/a770b5d1067488c7fd58dcf12b9fc07e6b3b91f0 😂 

for https://github.com/pingcap/dm/pull/1626#discussion_r619805000, I think if we support "delete them by prefix task directly" by delete their locks by prefix task directly. Not sure if there's such a senario

Reply:
Maybe a task name with '-', I'm afraid of mis-deleting them.